pkg/command/client.go

```diff
@@ -39,12 +37,21 @@ func (c *Client) SyncDashboard(ctx context.Context, uid string, queriesDir strin
 		queriesDirAbs = filepath.Join(wd, queriesDir)
 	}
 
-	queries := map[string]string{}
+	queries := QueryManager{}
 	if err := filepath.Walk(queriesDir, func(path string, info os.FileInfo, err error) error {
-		if info.IsDir() || filepath.Ext(path) != ".sql" {
+		if info.IsDir() {
 			return nil
 		}
-		queries[strings.TrimLeft(strings.ReplaceAll(path, queriesDirAbs, ""), "/")] = path
+
+		if filepath.Ext(path) == ".sql" || filepath.Ext(path) == ".promql" {
+			query, err := NewQueryFromFile(path)
+			if err != nil {
+				log.Printf("NewQueryFromFile: %s", err)
```

Comment:
Errors can be bubbled up instead of logged to sdtout.
If you need to decorate them, you can wrap it with **%w**, eg;
```go
return fmt.Errorf("NewQueryFromFile %s: %w", path, err)
```